Dockerfile.ios

```diff
@@ -1,30 +1,31 @@
 ARG img_version
 FROM godot-osx:${img_version}
 
-ENV IOS_SDK=16.1
+ENV IOS_SDK=16.4
 
 RUN dnf -y install --setopt=install_weak_deps=False \
       automake autoconf gcc gcc-c++ gcc-objc gcc-objc++ cmake libicu-devel libtool libxml2-devel openssl-devel perl python yasm && \
-    git clone --progress https://github.com/tpoechtrager/cctools-port.git && \
+    git clone --progress https://github.com/tpoechtrager/cctools-port && \
     cd /root/cctools-port && \
-    git checkout 11c93763d7e7ce7305163341d08052374e4712de && \
+    git checkout f28fb5e9c31efd3d0552afcce2d2c03cae25c1ca && \
+    curl -LO https://github.com/tpoechtrager/cctools-port/pull/136.patch && \
+    git apply 136.patch && \
     # arm64 device
     usage_examples/ios_toolchain/build.sh /root/files/iPhoneOS${IOS_SDK}.sdk.tar.xz arm64 && \
     mkdir -p /root/ioscross/arm64 && \
     mv usage_examples/ios_toolchain/target/* /root/ioscross/arm64 && \
     mkdir /root/ioscross/arm64/usr && \
     ln -s /root/ioscross/arm64/bin /root/ioscross/arm64/usr/bin && \
+    # Prepare for simulator builds
+    sed -i '/WRAPPER_SDKDIR/s/iPhoneOS/iPhoneSimulator/' usage_examples/ios_toolchain/build.sh && \
     # arm64 simulator
-    # Disabled for now as it doesn't work with cctools-port and current LLVM:
-    # Cf. https://github.com/tpoechtrager/cctools-port/issues/102
-    #sed -i 's/miphoneos-version-min/mios-simulator-version-min/g' usage_examples/ios_toolchain/wrapper.c && \
```

Comment:
For future reference, we discussed removing this with @bruvzg on RC:
```
<Akien> BTW, were our simulator builds a bit wonky due to this file being commented out, even though it seems relevant for both arm64_sim and x86_64_sim? https://github.com/godotengine/build-containers/blob/main/Dockerfile.ios#L20
<bruvzg> It seems to be for signing, which is irrelevant since we build static libs only. x86_64 simulator build were functional.
<bruvzg> And the rest of the args it's setting should be already set by Scons.
<Akien> Should I remove that line then to simplify the script?
<bruvzg> Probably, it was never used.
```